state-chain/pallets/cf-environment/src/tests.rs

```diff
@@ -352,3 +356,114 @@ fn test_consolidation_change_amount() {
 	)
 	.is_none());
 }
+
+#[test]
+fn test_sol_nonces_and_accounts_usage() {
+	new_test_ext().execute_with(|| {
+		SolanaAvailableNonceAccounts::<Test>::set(vec![
+			(SolAddress([1; 32]), SolHash([10; 32])),
+			(SolAddress([2; 32]), SolHash([20; 32])),
+			(SolAddress([3; 32]), SolHash([30; 32])),
+			(SolAddress([4; 32]), SolHash([40; 32])),
+			(SolAddress([5; 32]), SolHash([50; 32])),
+		]);
+
+		// Use one nonce
+		let (account1, nonce1) = Environment::get_sol_nonce_and_account().unwrap();
+		assert_eq!((account1, nonce1), (SolAddress([5; 32]), SolHash([50; 32])));
+		assert_eq!(SolanaUnAvailableNonceAccounts::<Test>::get(account1).unwrap(), nonce1);
+		assert_eq!(
+			SolanaUnAvailableNonceAccounts::<Test>::iter_keys().collect::<Vec<_>>().len(),
+			1
+		);
+
+		// use second nonce
+		let (account2, nonce2) = Environment::get_sol_nonce_and_account().unwrap();
+		assert_eq!((account2, nonce2), (SolAddress([4; 32]), SolHash([40; 32])));
+		assert_eq!(SolanaUnAvailableNonceAccounts::<Test>::get(account2).unwrap(), nonce2);
+		assert_eq!(
+			SolanaUnAvailableNonceAccounts::<Test>::iter_keys().collect::<Vec<_>>().len(),
+			2
+		);
+
+		// put back the first nonce
+		Environment::update_sol_nonce(RuntimeOrigin::root(), account1, nonce1).unwrap();
```

Comment:
It would be better to add a new *different* nonce than the initial one instead of adding the same on back again.